diff -ruwN _deps/xpdf-3.03/xpdf/GlobalParams.cc xpdf-3.03/xpdf/GlobalParams.cc
--- _deps/xpdf-3.03/xpdf/GlobalParams.cc	2011-08-15 23:08:53.000000000 +0200
+++ xpdf-3.03/xpdf/GlobalParams.cc	2011-10-16 22:33:46.513693391 +0200
@@ -85,20 +85,20 @@
   const char *t1FileName;
   const char *ttFileName;
 } displayFontTab[] = {
-  {"Courier",               "n022003l.pfb", "cour.ttf"},
-  {"Courier-Bold",          "n022004l.pfb", "courbd.ttf"},
-  {"Courier-BoldOblique",   "n022024l.pfb", "courbi.ttf"},
-  {"Courier-Oblique",       "n022023l.pfb", "couri.ttf"},
-  {"Helvetica",             "n019003l.pfb", "arial.ttf"},
-  {"Helvetica-Bold",        "n019004l.pfb", "arialbd.ttf"},
-  {"Helvetica-BoldOblique", "n019024l.pfb", "arialbi.ttf"},
-  {"Helvetica-Oblique",     "n019023l.pfb", "ariali.ttf"},
-  {"Symbol",                "s050000l.pfb", NULL},
-  {"Times-Bold",            "n021004l.pfb", "timesbd.ttf"},
-  {"Times-BoldItalic",      "n021024l.pfb", "timesbi.ttf"},
-  {"Times-Italic",          "n021023l.pfb", "timesi.ttf"},
-  {"Times-Roman",           "n021003l.pfb", "times.ttf"},
-  {"ZapfDingbats",          "d050000l.pfb", NULL},
+  {"Courier",               "fonts\\n022003l.pfb", "cour.ttf"},
+  {"Courier-Bold",          "fonts\\n022004l.pfb", "courbd.ttf"},
+  {"Courier-BoldOblique",   "fonts\\n022024l.pfb", "courbi.ttf"},
+  {"Courier-Oblique",       "fonts\\n022023l.pfb", "couri.ttf"},
+  {"Helvetica",             "fonts\\n019003l.pfb", "arial.ttf"},
+  {"Helvetica-Bold",        "fonts\\n019004l.pfb", "arialbd.ttf"},
+  {"Helvetica-BoldOblique", "fonts\\n019024l.pfb", "arialbi.ttf"},
+  {"Helvetica-Oblique",     "fonts\\n019023l.pfb", "ariali.ttf"},
+  {"Symbol",                "fonts\\s050000l.pfb", NULL},
+  {"Times-Bold",            "fonts\\n021004l.pfb", "timesbd.ttf"},
+  {"Times-BoldItalic",      "fonts\\n021024l.pfb", "timesbi.ttf"},
+  {"Times-Italic",          "fonts\\n021023l.pfb", "timesi.ttf"},
+  {"Times-Roman",           "fonts\\n021003l.pfb", "times.ttf"},
+  {"ZapfDingbats",          "fonts\\d050000l.pfb", NULL},
   {NULL}
 };

@@ -110,11 +110,13 @@
 };
 #else
 static const char *displayFontDirs[] = {
+  /*
   "/usr/share/ghostscript/fonts",
   "/usr/local/share/ghostscript/fonts",
   "/usr/share/fonts/default/Type1",
   "/usr/share/fonts/default/ghostscript",
   "/usr/share/fonts/type1/gsfonts",
+  */
   NULL
 };
 #endif
